src/Microsoft.SqlTools.ServiceLayer/Connection/ConnectionService.cs

```diff
@@ -578,6 +578,8 @@ private ConnectionCompleteParams GetConnectionCompleteParams(string connectionTy
                     ? reliableConnection.GetUnderlyingConnection()
                     : connection;
 
+                response.SPID = (underlyingConnection as SqlConnection).ServerProcessId;

```

Comment:
@cheenamalhotra Does the SPID still change when silent reconnections happen (which if I remember correctly was something that wasn't uncommon for Azure DBs)? Or is this guaranteed to stay the same until the user disconnects?

@Matteo-T @shueybubbles It looks like [SSMS refreshes the SPID on every query execution](https://msdata.visualstudio.com/SQLToolsAndLibraries/_git/SSMS?path=/Sql/ssms/core/SqlEditors/src/SqlScriptEditorControl.cs&version=GBmain&line=2909&lineEnd=2910&lineStartColumn=1&lineEndColumn=1&lineStyle=plain&_a=contents) - is the above why or is there anything else that you've run in to with ensuring this is kept up to date?

Reply:
That is true, SPID is subject to change if driver reconnects internally on query execution and STS may not send any connection complete events.

So ideally, we should be sending/reading SPID from `query/complete` instead of `connection/complete` as SPID is reflective of active connection used by the query editor, which can change internally at driver level.

Reply:
Just tested, SSMS doesn't update it's title everytime query executes.. which means it can lead to mis-information:

<img src="https://github.com/microsoft/sqltoolsservice/assets/13396919/acf9ae0b-7c74-4b93-8a32-e14665c4151a" width=300 />

If we decide to show this on titles, we need to be certain that query editor title will be refreshed on every query execution. But is it possible to do so without incurring UI/editor flickering on every query execution?

Reply:
we should probably remove the SPID from the title. The updated one is on the status bar. People would thank us for shortening the title text length!